### ISSUE TITLE
Propagate exit status code of plugin during installation with `export CF_CLI_EXPERIMENTAL=true`

### ISSUE THREAD

**Issue by glyn** [cloudfoundry/cli#1154]
The usecase is a plugin built with a string version number which parses the version number when the plugin is installed into `cf` and fails with a non-zero exit status code if the version number is malformed.

With `export CF_CLI_EXPERIMENTAL=false` the edge binary behaves the same as the release binary: the exit status code (64 was the value I chose FWIW) is returned from the `cf install-plugin` command. With `export CF_CLI_EXPERIMENTAL=true` an exit status code of 1 is returned from `cf install-plugin`.

Propagating the exit status code makes it easier to determine the reason for the installation failure. For instance, the plugin can _document_ that if the plugin was built with a malformed version, `cf install-plugin` will fail with exit status code 64.

So, the _current_ behaviour is desired, but this issue is to ensure the same behaviour under `export CF_CLI_EXPERIMENTAL=true`.

**Comment by mmb** [comment]
We are unable to reproduce the non-experimental code exiting with other than 1 regardless of what the 
plugin binary exits with.

We made a plugin that exits 99 (which causes the output "exit status 99") but cf still exits 1.

```
cf install-plugin configurable_plugin -f

Installing plugin configurable_plugin...
FAILED
exit status 99

$ echo $?
1
```

**Comment by glyn** [comment]
@mmb Apologies for the confusion. I am seeing the same behaviour and I should have written this issue in terms of the printing of the exit status message on the terminal. The usecase and principle is the same though - it's much easer to sort out the problem if the exit status code from the plugin is made available to the user. Let me rehash the issue description:

With `export CF_CLI_EXPERIMENTAL=false` the edge binary behaves the same as the release binary: the exit status code (64 was the value I chose FWIW) is printed by the `cf install-plugin` command. With `export CF_CLI_EXPERIMENTAL=true` exit status code is not printed by `cf install-plugin`.

Printing the exit status code makes it easier to determine the reason for the installation failure. For instance, the plugin can document that if the plugin was built with a malformed version, cf install-plugin will fail and print "exit status 64".

**Comment by mmb** [comment]
The "exit status 64" is printed by golang and not the CLI. With the experimental CLI code, either it is not being printed or it is being lost somehow.

Other messages printed to stdout and stderr by the plugin do show up when CF_CLI_EXPERIMENTAL=true. We will do more investigation into why the exit status message doesn't show up.

**Comment by glyn** [comment]
I'd be surprised if "exit status 64" is actually being printed by golang. Isn't it more likely that this is the text of an `error` which golang is returning from attempting to run the plugin and that the CLI is sometimes printing the text of the error and sometimes not? See the `String()` method of `ProcessState`, specifically [this line of code](https://github.com/golang/go/blob/2f73efa97136a6d4a602a94e87d2a948240e7e8a/src/os/exec_posix.go#L92).

**Comment by mmb** [comment]
I was testing with a main that does only `os.Exit(99)` and running it with `go run main.go`, which does output the exit status. You are right when you run a compiled binary it doesn't print the exit status.

I think these lines are where we lose the error:

https://github.com/cloudfoundry/cli/blob/master/actor/pluginaction/install.go#L110-L113

We could print the error here.

**Comment by glyn** [comment]
@mmb. Thanks. Printing the error would be most helpful.

**Comment by dkoper** [comment]
Addressed in 6.27.0. Thanks for reporting.
https://github.com/cloudfoundry/cli#downloads
